R/module_bookmark_manager.R

```diff
@@ -6,196 +6,129 @@
+              function(x) if (isTRUE(x)) "\u2705" else "\u274C",
+              how = "replace"
+            )
+            tags$div(
+              tags$p(
+                icon("fas fa-exclamation-triangle"),
+                "Some modules will not be restored when using this bookmark.",
+                "Check the list below to see which modules are not bookmarkable.",
+                class = "text-warning"
+              ),
+              tags$pre(yaml::as.yaml(bkmb_summary))
+            )
+          }
+        )
       }
-    })
 
-    ns <- session$ns
-
-    # Track input states ----
-    bookmark_history <- reactiveVal({
-      # Restore directly from bookmarked state, if applicable.
-      restoreValue(ns("bookmark_history"), list())
-    })
-
-    # Bookmark current input state - name bookmark. ----
-    observeEvent(input$bookmark_add, {
-      logger::log_trace("bookmark_manager_srv: bookmark_add button clicked")
       showModal(
         modalDialog(
-          textInput(ns("bookmark_name"), "Name the bookmark", width = "100%", placeholder = "Meaningful, unique name"),
-          footer = tagList(
-            actionButton(ns("bookmark_accept"), label = "Accept", icon = icon("thumbs-up")),
-            modalButton(label = "Cancel", icon = icon("thumbs-down"))
-          ),
-          size = "s"
+          title = "Bookmarked teal app url",
+          modal_content,
+          easyClose = TRUE
         )
       )
     })
 
-    # Initiate bookmarking with normal action button b/c `bookmarkButton` may not work on Windows.
-    observeEvent(input$bookmark_accept, {
-      app_session$doBookmark()
+    # manually trigger bookmarking because of the problems reported with bookmarkButton in teal
+    observeEvent(input$do_bookmark, {
+      logger::log_trace("bookmark_manager_srv@1 do_bookmark module clicked.")
+      session$doBookmark()
     })
-
-    # Create UI elements and server logic for the bookmark table ----
-    # Divs are tracked for a slight speed margin.
-    divs <- reactiveValues()
-
-    observeEvent(bookmark_history(), {
-      logger::log_trace("bookmark_manager_srv: bookmark history changed, updating bookmark list")
-      lapply(names(bookmark_history()), function(s) {
-        id_rowme <- sprintf("rowme_%s", make.names(s))
-
-        # Create a row for the bookmark table.
-        if (!is.element(id_rowme, names(divs))) {
-          divs[[id_rowme]] <- div(
-            class = "manager_table_row",
-            a(h5(s), title = "go to bookmark", href = bookmark_history()[[s]], target = "blank")
-          )
-        }
-      })
-    })
-
-    # Create table to display list of bookmarks and their actions ----
-    output$bookmark_list <- renderUI({
-      rows <- rev(reactiveValuesToList(divs))
-      if (length(rows) == 0L) {
-        div(
-          class = "manager_placeholder",
-          "Bookmarks will appear here."
-        )
-      } else {
-        rows
-      }
-    })
-
-    bookmark_history
```

Comment:
Maybe keep a return value for the sake of testing? If yes, rename properly here and in `wunder_bar_srv` accordingly.